recipes/aaf/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class AafConan(ConanFile):
+    name = "aaf"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://sourceforge.net/projects/aaf/"
+    description = "A  cross-platform SDK for AAF. AAF is a metadata management system and file format for use in professional multimedia creation and authoring."
+    topics = ("aaf", "multimedia", "crossplatform")
+    license = "AAFSDKPSL-2.0"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio" and self.settings.compiler.runtime in ("MT", "MTd"):
+            raise ConanInvalidConfiguration("Static runtime not supported")
+        if self.settings.os in ("FreeBSD", "Linux"):
+            raise ConanInvalidConfiguration("Linux is not (yet) supported by this recipe")
```

Comment:
When trying to build AAF on the `gcc8` docker image I have this issue:

```
/home/conan/clone/aaf/ref-impl/src/impl/aafErr.cpp: In function 'const char* aafGetErrorString(aafErr_t)':
/home/conan/clone/aaf/ref-impl/src/impl/aafErr.cpp:181:57: error: ISO C++ forbids comparison between pointer and integer [-fpermissive]
  300 && code >= 0 && localErrorStrings[0] != '\0')
                                              ^~~~

ref-impl/src/impl/CMakeFiles/Impl.dir/build.make:94: recipe for target 'ref-impl/src/impl/CMakeFiles/Impl.dir/aafErr.cpp.o' failed
make[2]: *** [ref-impl/src/impl/CMakeFiles/Impl.dir/aafErr.cpp.o] Error 1
CMakeFiles/Makefile2:623: recipe for target 'ref-impl/src/impl/CMakeFiles/Impl.dir/all' failed
make[1]: *** [ref-impl/src/impl/CMakeFiles/Impl.dir/all] Error 2
Makefile:102: recipe for target 'all' failed
make: *** [all] Error 2
```

I ignore if it is possible to silent this warning.

Reply:
that might be not that simple. it could be that make files hard-code `-Werror` (or `-Werror=<something>`) somewhere.
you may try to pass something like `add_definitions(-Wno-error)` from your cmake wrapper, but it could take least priority.
so it's possible there is no easy way to avoid it besides patching some make files...
in general, we do not require you to support all the configurations - if something is tricky to get it work, raising `ConanInvalidConfiguration` is more than acceptable and welcomed.
once someone needs an actual build of the tricky configuration, he may submit PR to improve the situation.